dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/EnrollmentService.java

```diff
@@ -39,12 +40,40 @@
 import org.hisp.dhis.tracker.PageParams;
 
 public interface EnrollmentService {
+
+  /**
+   * Finds the enrollment that matches the given {@code UID} based on the privileges of the
+   * currently authenticated user. Returns an {@link Optional} indicating whether the enrollment was
+   * found.
+   *
+   * @return an {@link Optional} containing the enrollment if found, or an empty {@link Optional} if
+   *     not
+   */
   @Nonnull
-  Enrollment getEnrollment(UID uid) throws ForbiddenException, NotFoundException;
+  Optional<Enrollment> findEnrollment(@Nonnull UID uid);
```

Comment:
I see it's only used in tests. Should it be moved to a test helper class instead?

Reply:
I think we agreed that this `get` and `find` are the 2 different ways to retrieve entities, even if it is not (yet) used it can make sense that our services are exposing such a method.